scripts/translations/download-locales.js

```
@@ -0,0 +1,195 @@
+/**
+ * @file
+ * This file is part of the HostlistsRegistry translation system.
+ * It handles the download of translation files (tags, filters, groups) from the translation service.
+ *
+ * The script downloads locale files from the translation service, converts them to the required format,
+ * and saves them to the appropriate locations in the repository.
+ */
+
+import fs from 'fs';
+import path from 'path';
+import axios from 'axios';
+import { logger } from '../helpers/logger.js';
+import { converter } from './converter.js';
+
+import {
+    LOCALES_DOWNLOAD_URL,
+    LOCALES_DIR,
+    TEMP_CONVERTED_FILE,
+    TEMP_MESSAGES_FILE
+} from './locales-constants.js'
+
+/**
+ * Get locales from .twosky.json configuration
+ * 
+ * @returns {string[]} Array of locale codes
+ */
+function getLocalesFromConfig() {
+    const twoskyPath = path.resolve(__dirname, '.twosky.json');
+    
+    try {
+        const twoskyConfig = JSON.parse(fs.readFileSync(twoskyPath, 'utf8'));
+        if (
+            Array.isArray(twoskyConfig)
+            && twoskyConfig.length > 0
+            && twoskyConfig[0].languages
+        ) {
+            return Object.keys(twoskyConfig[0].languages);
+        }
+    } catch (error) {
+        logger.error(`Error reading .twosky.json: ${error.message}`);
+    }
+
+    // Fallback to default locales if config can't be read
+    return [
```

let's rely on `.twosky.json` and its syntax validity always, and throw error instead of fallback